pkg/virt-operator/install-strategy/create.go

```diff
@@ -1275,6 +1292,185 @@ func generateServicePatch(kv *v1.KubeVirt,
+
+		apiService.Spec.CABundle = caBundle
+
+		obj, exists, _ := stores.APIServiceCache.Get(apiService)
+		// since these objects was in the past unmanaged, reconcile and pick it up if it exists
+		if !exists {
+			cachedAPIService, err = aggregatorClient.Get(apiService.Name, metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				exists = false
+			} else if err != nil {
+				return err
+			} else {
+				exists = true
+			}
+		} else if exists {
+			cachedAPIService = obj.(*v1beta1.APIService)
+		}
+
+		certsMatch := true
+		if exists {
+			if !reflect.DeepEqual(apiService.Spec.CABundle, cachedAPIService.Spec.CABundle) {
+				certsMatch = false
+			}
+		}
+
+		injectOperatorMetadata(kv, &apiService.ObjectMeta, version, imageRegistry, id)
+		if !exists {
+			expectations.APIService.RaiseExpectations(kvkey, 1, 0)
+			_, err := aggregatorClient.Create(apiService)
+			if err != nil {
+				expectations.APIService.LowerExpectations(kvkey, 1, 0)
+				return fmt.Errorf("unable to create apiservice %+v: %v", apiService, err)
+			}
+		} else {
+			if !objectMatchesVersion(&cachedAPIService.ObjectMeta, version, imageRegistry, id) || !certsMatch {
+				// Patch if old version
+				var ops []string
+
+				// Add Labels and Annotations Patches
+				labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&apiService.ObjectMeta)
+				if err != nil {
+					return err
+				}
+				ops = append(ops, labelAnnotationPatch...)
+
+				// Add Spec Patch
+				spec, err := json.Marshal(apiService.Spec)
+				if err != nil {
+					return err
+				}
+				ops = append(ops, fmt.Sprintf(`{ "op": "replace", "path": "/spec", "value": %s }`, string(spec)))
+
+				_, err = aggregatorClient.Patch(apiService.Name, types.JSONPatchType, generatePatchBytes(ops))
+				if err != nil {
+					return fmt.Errorf("unable to patch apiservice %+v: %v", apiService, err)
+				}
+				log.Log.V(2).Infof("apiservice %v updated", apiService.GetName())
+
+			} else {
+				log.Log.V(4).Infof("apiservice %v is up-to-date", apiService.GetName())
+			}
+		}
+	}
+	return nil
+}
+
+func createOrUpdateMutatingWebhookConfigurations(kv *v1.KubeVirt,
+	targetStrategy *InstallStrategy,
+	stores util.Stores,
+	clientset kubecli.KubevirtClient,
+	expectations *util.Expectations,
+	caBundle []byte,
+) error {
+
+	version := kv.Status.TargetKubeVirtVersion
+	imageRegistry := kv.Status.TargetKubeVirtRegistry
+	id := kv.Status.TargetDeploymentID
+
+	kvkey, err := controller.KeyFunc(kv)
+	if err != nil {
+		return err
+	}
+
+	for _, webhook := range targetStrategy.mutatingWebhookConfigurations {
+		var cachedWebhook *admissionregistrationv1beta1.MutatingWebhookConfiguration
+		webhook = webhook.DeepCopy()
+
+		if len(webhook.Webhooks) > 0 {
+			for i, _ := range webhook.Webhooks {
+				webhook.Webhooks[i].ClientConfig.CABundle = caBundle
+			}
+		}
+
+		obj, exists, _ := stores.MutatingWebhookCache.Get(webhook)
+		// since these objects was in the past unmanaged, reconcile and pick it up if it exists
```

Comment:
i guess our update functional tests will ensure all this works like expected now that we're transfering ownership of these objects to the operator

Reply:
yes. Tests are there now for this.